### ISSUE TITLE
Wrong song picture loaded

### ISSUE THREAD

**Issue by xhuytox** [Alaanor/beatlist#5]
This isn't big deal; only aesthetic. To reproduce, just type into the search really quickly. Preferably, type in a word that a lot of song names. While it tries to live-filter while the typing is happening, sometimes the wrong song pictures get associated to the final results.

**Comment by Alaanor** [comment]
Yeah, I noticed this bug already, you're right, this happens when we type fast. I'm pretty sure it's because of the order of the async event. At each letter, it does the filter and asks for a new image. And sometime the previous search was slower than the new one, resulting in getting the wrong image.

It is on my task list. Thanks for the report.

**Comment by DanP27** [comment]
I am also experiencing this bug - is there a fix planned?

**Comment by Alaanor** [comment]
@DanP27 Yes, as you can see the commit referenced above (68e7987823438d778dcaaf0741515c707a11988b) is fixing the issue, however, it's planned for the next major release. (v1.1, which I'm working on it since months on my free time 👌 ) 
You can see the work done (and try it if you know the tools) on the [feature-beatsaver-integration](https://github.com/Alaanor/beatlist/tree/feature-beatsaver-integration) branch.
